x/supplier/client/cli/stake_options_reader.go

```diff
@@ -0,0 +1,157 @@
+}
+
+// ServiceEndpoint is the structure describing a single service endpoint in the stake config file
+type ServiceEndpoint struct {
+	Url     string            `yaml:"url"`
+	RPCType string            `yaml:"rpc_type"`
+	Config  map[string]string `yaml:"config"`
+}
+
+// parseStakeConfig parses the stake config file into a parsedStakeConfig
+func parseStakeConfig(configContent []byte) (*ParsedStakeConfig, error) {
+	var (
+		stakeConfig *StakeConfig
+		err         error
+	)
+
+	// Unmarshal the stake config file into a stakeConfig
+	if err = yaml.Unmarshal(configContent, &stakeConfig); err != nil {
+		return nil, err
+	}
+
+	// Prepare the parsedStakeConfig
+	parsedStakeConfig := &ParsedStakeConfig{}
+
+	// Parse the stake amount and assign it to the parsedStakeConfig
+	parsedStakeConfig.stake, err = sdk.ParseCoinNormalized(stakeConfig.Stake)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the services slice
+	var services []*sharedtypes.SupplierServiceConfig
+
+	// Populate the services slice
+	for _, svc := range stakeConfig.Services {
+		// Validate the serviceId
+		// TODO_TECH_DEBT: This should be validated against some governance state
+		// defining the network's supported services
+		if svc.ServiceId == "" {
+			return nil, fmt.Errorf("invalid serviceId")
+		}
+
+		// Create a supplied service config with the serviceId
+		service := &sharedtypes.SupplierServiceConfig{
+			Service:   &sharedtypes.Service{Id: svc.ServiceId},
+			Endpoints: []*sharedtypes.SupplierEndpoint{},
+		}
+
+		// Iterate over the service endpoints and add their parsed representation to the supplied service config
+		for _, endpoint := range svc.Endpoints {
+			parsedEndpointEntry, err := parseEndpointEntry(endpoint)
+			if err != nil {
+				return nil, err
+			}
+			service.Endpoints = append(service.Endpoints, parsedEndpointEntry)
+		}
+		services = append(services, service)
+	}
+
+	parsedStakeConfig.services = services
+
+	return parsedStakeConfig, nil
+}
+
+func parseEndpointEntry(endpoint ServiceEndpoint) (*sharedtypes.SupplierEndpoint, error) {
+	endpointEntry := &sharedtypes.SupplierEndpoint{}
+
+	// Endpoint URL
+	endpointUrl, err := validateEndpointURL(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	endpointEntry.Url = endpointUrl
+
+	// Endpoint config
+	endpointEntry.Configs = parseEndpointConfigs(endpoint)
+
+	// Endpoint RPC type
+	endpointEntry.RpcType = parseEndpointRPCType(endpoint)
+
+	return endpointEntry, nil
+}
+
+func validateEndpointURL(endpoint ServiceEndpoint) (string, error) {
+	// Validate the endpoint URL
+	if _, err := url.Parse(endpoint.Url); err != nil {
+		return "", err
+	}
+
+	return endpoint.Url, nil
+}
+
+func parseEndpointConfigs(endpoint ServiceEndpoint) []*sharedtypes.ConfigOption {
+	// Prepare the endpoint configs slice
+	endpointConfigs := []*sharedtypes.ConfigOption{}
+
+	// If we have an endpoint config entry, parse it into a slice of ConfigOption
+	if endpoint.Config != nil {
```

Comment:
`if endpoint.Config == nil { return }`

Then you don't need to indent all the business logic